lua/nvim-treesitter/shell_command_selectors.lua

```diff
@@ -121,12 +121,16 @@ end
 
 function M.select_download_commands(repo, project_name, cache_folder, revision)
 
+  local has_tar = vim.fn.executable('tar') == 1
+  local has_curl = vim.fn.executable('curl') == 1
   local is_github = repo.url:find("github.com", 1, true)
   local is_gitlab = repo.url:find("gitlab.com", 1, true)
+  local is_windows = fn.has('win32') == 1
 
-  if vim.fn.executable('tar') == 1 and vim.fn.executable('curl') == 1 and (is_github or is_gitlab) then
+  revision = revision or repo.branch or "master"
+
+  if has_tar and has_curl and (is_github or is_gitlab) and not is_windows then
```

Comment:
Could this logic be a bit more encapsulated? It's a bit difficult to understand when the windows conditions is added additionally. Maybe just introduce for `has_tar and has_curl and (is_github or is_gitlab)` a new variable, e.g. `can_use_tar`?

`if can_use_tar and not is_windows` is a bit more compact.

Reply:
I've changed this in my latest commit according to your suggestions. Thanks. 👍 
I've added `is_github_or_gitlab`, though.